### PR TITLE
refactor(hub): new rss3 uri parser and test cases

### PR DESCRIPTION
**Pull request by kallydev**

### CODE REVIEW

**Comment by kallydev** [comment]
I have classified `Instance` into two types, `PlatformInstance` and `NetworkInstance`.

Please don't merge this branch yet, it's still being refined.

File: hub/pkg/rss3uri/uri.go [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+	ErrInvalidPlatform = errors.New("invalid platform")
+	ErrInvalidNetwork  = errors.New("invalid network")
+)
+
+var (
+	_ fmt.Stringer = &URI{}
+)
+
+type URI struct {
+	Instance Instance `json:"instance"`
```
Comment (Songkeys) [inline hub/pkg/rss3uri/uri.go:26]: this looks reasonable but what more fields do you think would here be? i think gin's router bindings could help us parse the paths and query strings?

**Comment by Songkeys** [comment]
Thank you.

File: hub/pkg/rss3uri/uri.go (hunk 1 above)
Comment (kallydev) [reply hub/pkg/rss3uri/uri.go:26]: I'm mostly using [echo](https://github.com/labstack/echo), if I find a more suitable solution, I'll implement it.

**Pull request merged**